packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts

```diff
@@ -4,12 +4,49 @@ import { IEventSourceDlq } from './dlq';
 import { IFunction } from './function-base';
 import { CfnEventSourceMapping } from './lambda.generated';
 
+/**
+ * Specific settings like the authentication protocol or the VPC components to secure access to your event source.
+ */
+export interface SourceAccessConfiguration {
+  /**
+   * The type of authentication protocol or the VPC components for your event source. For example: "SASL_SCRAM_512_AUTH".
+   * Valid values are: BASIC_AUTH | VPC_SUBNET | VPC_SECURITY_GROUP | SASL_SCRAM_512_AUTH | SASL_SCRAM_256_AUTH
+   */
+  readonly type: string,
```

Comment:
Model this as an enum-like class.

Here are some examples - [UserPoolOperation](https://github.com/aws/aws-cdk/blob/22eb7e1816fa457cbbba5474aff08974f2793e2d/packages/%40aws-cdk/aws-cognito/lib/user-pool.ts#L149) or [Runtime](https://github.com/aws/aws-cdk/blob/22eb7e1816fa457cbbba5474aff08974f2793e2d/packages/%40aws-cdk/aws-lambda/lib/runtime.ts#L39)

Reply:
Done.